client/components/timezone/index.jsx

```diff
@@ -15,9 +16,32 @@ import QueryTimezones from 'components/data/query-timezones';
 import getRawOffsets from 'state/selectors/get-raw-offsets';
 import getTimezones from 'state/selectors/get-timezones';
 
+function maybeFindLinkedTimezone( timezone, timezones ) {
+	if ( ! timezones || timezone in timezones ) {
+		return timezone;
+	}
+	// The (guessed) timezone could not be found, so let's see if a linked timezone for this exists.
+	// Example: we don't have Asia/Calcutta but Asia/Kalkota, so we'll need to choose that.
+	timezone = reduce(
+		pickBy( flattenDepth( timezones, 2 ), isArray ),
+		function( mappedTimezone, option ) {
+			const [ value ] = option;
+			// This uses the internal linking array of moment-timezone since the links are not exposed otherwise.
+			const mapping = moment.tz._links[ value.toLowerCase().replace( /\//g, '_' ) ];
+			if ( mapping && moment.tz.zone( mapping ).name === mappedTimezone ) {
+				return value;
+			}
+			return mappedTimezone;
```

Comment:
Is this function supposed also to find and follow chains of multiple links? E.g., `A` links to `B` and `B` links to `C`, and the expected outcome is that we find a link from `A` to `C`?

Reply:
It's a side-effect of this implementation but usually it wouldn't. Maybe I should just do two for loops which would also make this more readable.

Reply:
If it's not intentional, then `find` is a better function than `reduce`. After it finds a match, `find` doesn't go through the rest of the list like `reduce` but breaks early.